### DISCUSSION TITLE
Yeni Örnek Oyun

### DISCUSSION THREAD

**Discussion by sezginkipel** [sezginkipel/Oyun-Programlama-Dersi#6] [General]
Örnek bir proje olarak, ingilizce kelime-görsel bulmaca tarzında bir oyun hazırladım. Aşağıdaki repodan inceleyebilirsiniz, kelime sayısını sadece örnek olması açısından düşük tuttum.

[Oyun Reposu](https://github.com/sezginkipel/Word-Runner)

**Comment by 198168Cf189168Cf** [comment]
örneğindeki kodlara erişmekte zorlanıyorum başka bir yolu varsa yardımcı olur musun yada direkt yazar mısın ?

**Reply by sezginkipel** [reply]
https://discord.gg/8ffsSQ4a

Discord'a gel oradan bakalım.

**Reply by 198168Cf189168Cf** [reply]
süresi geçmiş

**Comment by 198168Cf189168Cf** [comment]
oluşturduğum nesneleri dizinin içine atıp rastgele nasıl çağırabilirim ?

**Reply by sezginkipel** [reply]
Sürekli yeni kelimeler ekleyeceğin için dizi yerine liste ya da sözlük kullanabilirsin.

Ben şu şekilde bir yapı kullanmıştım örnek oyunda;

`public List<Sprite> objectSprites; // Kelimelerin spritelarını burada tutuyorum.`
`private Dictionary<string, Sprite> _wordToSpriteMapping; // Kelime ile kelimeye ait sprite değerini tutacak`

`_wordToSpriteMapping.Add("Apple", objectSprites[0]); // Sözlüğe "Apple" kelimesini ekledim ve 0 indexine sahip sprite ile eşleştirdim`

Tüm bunları yaptıktan sonra Rastgele bir sayı üretip bu sprite/kelime listesinden bir öğeyi seçiyorum.

Kelime ve sprite oluşturma işlemine ait kodlara aşağıdaki linkten erişebilirsin.

https://github.com/sezginkipel/Word-Runner/blob/master/Assets/Scripts/WordGenerator.cs
